pkg/clusterversion/cv.go

```diff
@@ -64,44 +68,67 @@ func (c *clusterVersionClient) GetClusterVersion() (*configv1.ClusterVersion, er
 	return cv, err
 }
 
-func (c *clusterVersionClient) EnsureDesiredVersion(uc *upgradev1alpha1.UpgradeConfig) (bool, error) {
+func (c *clusterVersionClient) EnsureDesiredConfig(uc *upgradev1alpha1.UpgradeConfig) (bool, error) {
 	clusterVersion, err := c.GetClusterVersion()
 	if err != nil {
 		return false, err
 	}
 
-	// Move the cluster to the same channel first
 	desired := uc.Spec.Desired
-	if clusterVersion.Spec.Channel != desired.Channel {
-		clusterVersion.Spec.Channel = desired.Channel
-		err = c.client.Update(context.TODO(), clusterVersion)
-		if err != nil {
-			return false, err
+
+	// Commence upgrade using version and channel
+	// Move the cluster to the same channel first
+	if !empty(desired.Channel) && !empty(desired.Version) {
+		if clusterVersion.Spec.Channel != desired.Channel {
+			logger.Info(fmt.Sprintf("Setting ClusterVersion to Channel %s Version %s", desired.Channel, desired.Version))
+			clusterVersion.Spec.Channel = desired.Channel
+			err = c.client.Update(context.TODO(), clusterVersion)
+			if err != nil {
+				return false, err
+			}
+
+			// Retrieve the updated version
+			clusterVersion, err = c.GetClusterVersion()
```

Comment:
what is the purpose for fetching the clusterversion here? do we need to verify the returned version also?

Reply:
This change was introduced in https://github.com/openshift/managed-upgrade-operator/pull/168/files#diff-d7acacac799c3b89ebeedaaf44a111ed1cde3e3e09c3f333f67d25aa213fa96e actually.. The diff seems to show this change as part of a new block though.. I think this is just intended to fetch the updated `ClusterVersion` resource and no need to verify the version yet since that's replaced in code below as such..

Reply:
ah, after a bit more reading, I think I got this, it is just re-write the current clusterversion value (after the update above) to the `clusterVersion` variable.